src/clincoded/static/components/provisional_classification/approval.js

```diff
@@ -124,19 +204,25 @@ const ClassificationApproval = module.exports.ClassificationApproval = createRea
             approvalReviewDate: this.props.provisional && this.props.provisional.approvalReviewDate ? this.props.provisional.approvalReviewDate : undefined,
             approvalComment: this.props.provisional && this.props.provisional.approvalComment ? this.props.provisional.approvalComment : undefined,
             classificationApprover: this.props.provisional && this.props.provisional.classificationApprover ? this.props.provisional.classificationApprover : undefined,
+            retainSelectedApprover: this.props.provisional && this.props.provisional.retainSelectedApprover ? this.props.provisional.retainSelectedApprover : null,
+            retainSelectedContributor: this.props.provisional && this.props.provisional.retainSelectedContributor ? this.props.provisional.retainSelectedContributor : null,
+            classificationContributors: this.props.provisional && this.props.provisional.classificationContributors ? this.props.provisional.classificationContributors : null,
+            additionalApprover: this.props.provisional && this.props.provisional.additionalApprover ? this.props.provisional.additionalApprover : null,
+            contributorComment: this.props.provisional && this.props.provisional.contributorComment ? this.props.provisional.contributorComment : null,
+            sopVersion: this.props.provisional && this.props.provisional.sopVersion ? this.props.provisional.sopVersion : null,
             isApprovalPreview: false
         });
     },
 
     /**
-     * Method to handle editing classificaiton approval form
+     * Method to handle editing classification approval form
      */
     handleEditApproval() {
-        this.setState({isApprovalPreview: false});
+        this.setState({ isApprovalPreview: false });
     },
 
     /**
-     * Method to handle submitting classificaiton approval form
+     * Method to handle submitting classification approval form
      */
     submitForm(e) {
```

Comment:
Honestly not too sure if the best practice here would be to allow users to retry if something fails here, or if we should keep the buttons disabled so they cannot keep trying to submit if it's failing. 

Users might not know to open the console and check for errors indicating the submit function failed, and I think grayed out buttons and the appearance that it is stuck as 'busy' might be better for identifying that. Although the option to enable Cancel Approval here may help, but users may just refresh their browser in that case anyway.  Let me know what you think.

Reply:
My answer to this got pasted twice below this comment, deleted it and now the originals gone. Hopefully you received it in the email thread sent by Github. Also looking into the commit history for this function, it looks like submitBusy was initially set to false like you recommend, but it's currently not there. I'll look into why it was removed to see if we should reintroduce it.

Reply:
No worries, I got the email. You bring up a fair point. In the case where the error is persistent, we don't want the user to repeatedly submit the form without any kind of feedback. However, there's also the case where a one-time error could happen (i.e. user's internet cuts out for a minute). The user would likely want to try submitting again. While this could be solved, like you said, by refreshing the browser, I think the user experience suffers a bit, especially if they have to re-enter the information for the approval.

I think ideally, we would have some sort of message that appears as feedback to the user, so we don't need to keep the buttons "busy" when the app isn't actually "busy".  This is probably not a viable solution for now, since this will take more development work (I'm actually working on this error message feature).

After thinking about it some more, since we don't have this feedback message in place, I think I agree that the buttons should stay disabled. After you've read this, you can go ahead and resolve this conversation.